packages/react/src/components/TimePicker/TimePicker.tsx

```diff
@@ -0,0 +1,150 @@
+import * as React from 'react';
+import { TimeConstants, addMinutes, formatTimeString, ceilMinuteToIncrement } from '@fluentui/date-time-utilities';
+import { ComboBox, IComboBox, IComboBoxOption } from '../../ComboBox';
+import { ITimePickerProps, ITimeRange } from './TimePicker.types';
+
+// Valid KeyChars for user input
+const KEYCHAR_0 = 48;
+const KEYCHAR_COLON = 58;
+const KEYCHAR_SPACE = 32;
+const KEYCHAR_A = 65;
+const KEYCHAR_M = 77;
+const KEYCHAR_P = 80;
+
+export const TimePicker = ({
+  label,
+  increments = 30,
+  showSeconds = false,
+  allowFreeform = true,
+  useHour12 = false,
+  timeRange = { start: -1, end: -1 },
+  onFormatDate,
+  onValidateUserInput,
+  ...rest
+}: ITimePickerProps) => {
+  const [selectedKey, setSelectedKey] = React.useState<string | number | undefined>(0);
+  const [userText, setUserText] = React.useState<string>('');
+  const [errorMessage, setErrorMessage] = React.useState<string>('');
+
+  const defaultTime = generateDefaultTime(increments, timeRange);
+  const optionsCount = getDropdownOptionsCount(increments, timeRange);
+  const timePickerOptions: IComboBoxOption[] = Array(optionsCount)
+    .fill(0)
+    .map((_, index) => {
+      const option = addMinutes(defaultTime, increments * index);
+      option.setSeconds(0);
+      return {
+        key: index,
+        text: onFormatDate ? onFormatDate(option) : `${formatTimeString(option, showSeconds, useHour12)}`,
+      };
+    });
+
+  const onChange = React.useCallback(
+    (event: React.FormEvent<IComboBox>, option?: IComboBoxOption, index?: number, value?: string): void => {
+      const validateUserInput = (userInput: string): string => {
+        let errorMessageToDisplay = '';
+        let regexString;
+        if (useHour12) {
+          regexString = showSeconds
+            ? /((1[0-2]|0?[1-9]):([0-5][0-9]):(?:[0-5]\d) ?([AaPp][Mm]))$/
+            : /((1[0-2]|0?[1-9]):([0-5][0-9]) ?([AaPp][Mm]))$/;
+        } else {
+          regexString = showSeconds
+            ? /([0-9]|0[0-9]|1[0-9]|2[0-3]):(?:[0-5]\d):(?:[0-5]\d)$/
+            : /([0-9]|0[0-9]|1[0-9]|2[0-3]):(?:[0-5]\d)$/;
+        }
+        if (!regexString.test(userInput)) {
+          const useHour12ErrorMessage = useHour12 ? '12-hour' : '24-hour';
+          showSeconds
+            ? (errorMessageToDisplay =
+                `TimePicker format must be valid and in the ${useHour12ErrorMessage} ` + `format hh:mm:ss A.`)
```

Comment:
Rather than hard-coding English strings, you should add a `strings` prop like DatePicker has.

Reply:
Resolved by providing `strings` prop.